lua/telescope/previewers/buffer_previewer.lua

```diff
@@ -1,60 +1,34 @@
-local debounce = require('telescope.debounce')
 local from_entry = require('telescope.from_entry')
 local path = require('telescope.path')
 local utils = require('telescope.utils')
 local putils = require('telescope.previewers.utils')
 local Previewer = require('telescope.previewers.previewer')
+local conf = require('telescope.config').values
 
 local pfiletype = require('plenary.filetype')
 
-local has_ts, _ = pcall(require, 'nvim-treesitter')
-local _, ts_highlight = pcall(require, 'nvim-treesitter.highlight')
-local _, ts_parsers = pcall(require, 'nvim-treesitter.parsers')
-
 local buf_delete = utils.buf_delete
-
 local defaulter = utils.make_default_callable
 
 local previewers = {}
 
-local previewer_ns = vim.api.nvim_create_namespace('telescope.previewers')
+local ns_previewer = vim.api.nvim_create_namespace('telescope.previewers')
 
-local file_maker_async = function(filepath, bufnr, bufname, callback)
-  local ft = pfiletype.detect(filepath)
+previewers.file_maker = function(filepath, bufnr, bufname, use_ft_detect, callback)
```

Comment:
Maybe it would be better to have any function more then 3 params to accept a dict to increase readability

Reply:
Nope :). All are required and need to be handled and that function is basically only public for an extension